src/diffpy/morph/morphs/morphsqueeze.py

```diff
@@ -0,0 +1,73 @@
+import numpy as np
+from numpy.polynomial import Polynomial
+from scipy.interpolate import CubicSpline
+
+from diffpy.morph.morphs.morph import LABEL_GR, LABEL_RA, Morph
+
+
+class MorphSqueeze(Morph):
+    """Apply a polynomial to squeeze the morph function. The morphed
+    data is returned on the same grid as the unmorphed data."""
+
+    # Define input output types
+    summary = "Squeeze morph by polynomial shift"
+    xinlabel = LABEL_RA
+    yinlabel = LABEL_GR
+    xoutlabel = LABEL_RA
+    youtlabel = LABEL_GR
+    parnames = ["squeeze"]
+
+    def morph(self, x_morph, y_morph, x_target, y_target):
+        """Squeeze the morph function.
+
+        This applies a polynomial to squeeze the morph non-linearly.
+
+        Configuration Variables
+        -----------------------
+        squeeze : list
+            The polynomial coefficients [a0, a1, ..., an] for the squeeze
+            function where the polynomial would be of the form
+            a0 + a1*x + a2*x^2 and so on.  The order of the polynomial is
+            determined by the length of the list.
+
+        Returns
+        -------
+            A tuple (x_morph_out, y_morph_out, x_target_out, y_target_out,
+            min_index, max_index) where the target values remain the same and
```

Comment:
we don't need to return min and max index.  These can be just updated in place.  Also, these names are not so intuitive to me. maybe something like `extrap_index_low` and `extrap_index_high`?    Let's define these and set them to `None` when `MorphSqueeze` is instantiated.